python/lsst/dax/imgserv/getimage/imageget.py

```diff
@@ -0,0 +1,625 @@
+            ss = SpanSet.fromShape(pix_r)
+            ss_width = ss.getBBox().getWidth()
+            src_image = self.cutout_from_nearest(ra, dec, ss_width, ss_width,
+                                                 "pixel", filt)
+            src_cutout = src_image.getMaskedImage()
+            circle_cutout = afwImage.MaskedImageF(src_cutout.getBBox())
+            spanset= SpanSet.fromShape(pix_r, Stencil.CIRCLE,
+                                       offset=src_cutout.getXY0() +
+                                              afwGeom.Extent2I(pix_r,
+                                                               pix_r))
+            spanset.copyMaskedImage(src_cutout, circle_cutout)
+            # make an Exposure cutout with WCS info
+            cutout = afwImage.ExposureF(circle_cutout,
+                                            afwImage.ExposureInfo(wcs))
+            return cutout
+        elif shape == "RANGE":
+            # convert the pair of (ra,dec) to bbox
+            ra1, ra2 = float(pos_items[1]), float(pos_items[2])
+            dec1, dec2 = float(pos_items[3]), float(pos_items[4])
+            box = afwGeom.Box2D(afwGeom.Point2D(ra1, dec1),
+                                  afwGeom.Point2D(ra2, dec2))
+            # convert from deg to arcsec
+            w = box.getWidth()*3600
+            h = box.getHeight()*3600
+            # compute the arithmetic center (ra, dec) of the range
+            ra = (ra1 + ra2) / 2
+            dec = (dec1 + dec2) / 2
+            cutout = self.cutout_from_nearest(ra, dec, w, h, "arcsec", filt)
+            return cutout
+        elif shape == "POLYGON":
+            vertices = []
+            pos_items.pop(0)
+            for long, lat in zip(pos_items[::2], pos_items[1::2]):
+                pt = afwGeom.Point2D(float(long), float(lat))
+                vertices.append(pt)
+            polygon = afwGeom.Polygon(vertices)
+            center = polygon.calculateCenter()
+            ra, dec = center.getX(), center.getY()
+            # afw limitation: can only return the bbox of the polygon
+            bbox = polygon.getBBox()
+            # convert from 'deg' to 'arcsec'
+            w = bbox.getWidth()*3600
+            h = bbox.getHeight()*3600
+            cutout = self.cutout_from_nearest(ra, dec, w, h, "arcsec", filt)
+            return cutout
+
+    @staticmethod
+    def cutout_from_exposure(src_image, ra, dec, width, height, unit="pixel"):
+        """ Get the Exposure cutout including wcs headers.
+
+        Parameters
+        ----------
+        src_image : `afwImage.Exposure`
+            the source image.
+        ra : `float`
+            in degrees.
+        dec : `float`
+            in degrees.
+        width : int
+            in pixels.
+        height : int
+            in pixels
+
+        Returns
+        -------
+        cutout : `afwImage.Exposure`
+
+        """
+        if unit == "arcsec":
```

Comment:
There's no else here, maybe instead of doing an if, you should throw an exception if unit isn't arcsec

Reply:
Actually, this is a bit more confusing, since if you pass in arc secs, then you're redefining what width and height are.  At least add unit to the parameters and say something about doing the conversion?  Seems even safer to just have a different function?